### PR TITLE
feat: add option to ignore cache

### PR DESCRIPTION
**Pull request by LironMShemen**
- Introduced a new environment variable to bypass cache.
- Updated README to include usage instructions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +49,13 @@ reset: () => void;
 perform: (steps: string | string[]) => Promise<any | any[]>;
 ```
 
+### **Additional Note**
+
+In addition to the operations history, Copilot maintains a cache. If you need to ignore the current cache for any reason (e.g., when adding an action to the API driver), you can set the environment variable process.env.COPILOT_OVERRIDE_CACHE to "true" before running your tests. This will ensure that the current cache is not taken into consideration.
+```bash
+process.env.COPILOT_OVERRIDE_CACHE = "1"
```
Comment (asafkorem) [inline README.md:56]: so both "1" and "true" are supported? are you sure?

Comment (asafkorem) [reply README.md:56]: also, it's not a bash script but JS script

Comment (asafkorem) [reply README.md:56]: replace it with `export COPILOT_OVERRIDE_CACHE=true`

File: src/actions/StepPerformer.ts [hunk 2]
```diff
@@ -71,7 +71,7 @@ export class StepPerformer {
     ): Promise<string> {
         const cacheKey = this.generateCacheKey(step, previous, viewHierarchy);
 
-        if (this.cache.has(cacheKey)) {
+        if (process.env.COPILOT_OVERRIDE_CACHE != "true" && this.cache.has(cacheKey)) {
```
Comment (asafkorem) [inline src/actions/StepPerformer.ts:74]: why `!=` and not `!==`? 🙂

Comment (asafkorem) [reply src/actions/StepPerformer.ts:74]: can you write a test for that? add it to `StepPerformer.test.ts`

**Review by asafkorem — CHANGES_REQUESTED** [review]
round 2

**Comment by LironMShemen** [comment]
I changed my code according to the comments. Mistakenly, I did the commit with amend.

**Comment by asafkorem** [comment]
Actually, squashing (or in this case - amending) commit changes after a code review is actually a great practice as it keeps the history clean and focused on the **final**, relevant changes. 

For larger reviews, I do prefer separate commits to show the specific fixes from the code review, and then once everything is approved, we can squash them into more meaningful, contained commits.

However, since that's a small change, I'm perfectly fine with the amend approach here, it keeps things clean and simple

File: src/actions/StepPerformer.test.ts [hunk 3]
```diff
@@ -244,4 +250,20 @@ describe('StepPerformer', () => {
...
+        const result = await stepPerformer.perform(intent);
+
+        expect(result).toBe('success');
+        // Should call runPrompt or createPrompt. Shouldn't use current cache, but override it
+        expect(mockPromptCreator.createPrompt).toHaveBeenCalled();
+        expect(mockPromptHandler.runPrompt).toHaveBeenCalled();
+        expect(mockCodeEvaluator.evaluate).toHaveBeenCalledWith('generated code', mockContext);
+        expect(fs.writeFileSync).toHaveBeenCalled(); // Need to save cache
+
+
```
Comment (asafkorem) [inline src/actions/StepPerformer.test.ts:267]:
```suggestion
```

File: src/actions/StepPerformer.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ import * as path from 'path';
 import * as crypto from 'crypto';
 import {extractCodeBlock} from '@/utils/extractCodeBlock';
 
+
```
Comment (asafkorem) [inline src/actions/StepPerformer.ts:10]:
```suggestion
```

**Review by asafkorem — CHANGES_REQUESTED** [review]

**Pull request merged**